fedbiomed/researcher/requests/_requests.py

```diff
@@ -0,0 +1,495 @@
+# This file is originally part of Fed-BioMed
+# SPDX-License-Identifier: Apache-2.0
+
+"""
+Implements the message exchanges from researcher to nodes
+"""
+
+import json
+import os
+import uuid
+import tempfile
+import threading
+from typing import Any, Dict, Callable, Union, List, Optional
+
+import tabulate
+from python_minifier import minify
+
+from fedbiomed.common.constants import MessageType
+from fedbiomed.common.logger import logger
+from fedbiomed.common.message import ResearcherMessages, SearchRequest, ErrorMessage, Message
+from fedbiomed.common.singleton import SingletonMeta
+from fedbiomed.common.training_plans import BaseTrainingPlan
+from fedbiomed.common.utils import import_class_object_from_file
+
+from fedbiomed.transport.server import GrpcServer
+from fedbiomed.transport.node_agent import NodeAgent, NodeActiveStatus
+
+from fedbiomed.researcher.environ import environ
+
+from ._policies import RequestPolicy, PolicyController, DiscardOnTimeout
+from ._status import RequestStatus, PolicyStatus
+
+# timeout in seconds for checking disconnection and node status changes
+REQUEST_STATUS_CHECK_TIMEOUT = 0.5
+
+
+class MessagesByNode(dict):
+    """Type to defined messages by node"""
+    pass
+
+
+class Request:
+
+    def __init__(
+        self,
+        message: Message,
+        node: NodeAgent,
+        request_id: Optional[str] = None,
+        sem_pending: threading.Semaphore = None
+    ) -> None:
+        """Single request for node
+
+        Args:
+            message: Message to send to the node
+            node: Node agent
+            request_id: unique ID of request
+            sem_pending: semaphore for signaling new pending reply
+        """
+        self.request_id = request_id if request_id else str(uuid.uuid4())
```

Comment:
hi @srcansiz 

Same remark as `FederatedRequests` : it would probably be nice to have private variables with getters when needed (and setter for `self.status` to allow policies to set it)

Reply:
`request_id` can be private but creating setter for `self.status` is not needed. It will just increase the complexity. These attribute are only used developers and I don't see any reason to implement setter to validate the inputs.

Reply:
Yes, you're right it's not needed for `self.status` as it's read/write.

Yet `status` `error` and `reply` are probably concerned, and maybe also `message` and `node`